apps/gitness/src/pages/execution/execution-details.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { useEffect, useState } from 'react'
+import { useParams } from 'react-router-dom'
+import { TypesStage, useFindExecutionQuery, useViewLogsQuery } from '@harnessio/code-service-client'
+import { Badge, Icon, ScrollArea, Separator, Text } from '@harnessio/canary'
+import {
+  Layout,
+  ExecutionTree,
+  ExecutionStatus,
+  StageExecution,
+  ContactCard,
+  convertExecutionToTree,
+  StageProps
+} from '@harnessio/playground'
+import { PathParams } from '../../RouteDefinitions'
+import { useGetRepoRef } from '../../framework/hooks/useGetRepoPath'
+import { ExecutionState } from '../../types'
+import { getDuration, timeAgoFromEpochTime } from '../pipeline-edit/utils/time-utils'
+
+const ExecutionLogs: React.FC = () => {
+  const { pipelineId, executionId } = useParams<PathParams>()
+  const repoRef = useGetRepoRef()
+  const [stage, setStage] = useState<TypesStage>()
+  const [stageNum, setStageNum] = useState<number>(1)
+  const [stepNum, setStepNum] = useState<number>(1)
+  const pipelineIdentifier = pipelineId || ''
+  const executionNum = executionId || ''
+
+  const { data: execution } = useFindExecutionQuery({
+    pipeline_identifier: pipelineIdentifier,
+    execution_number: executionNum,
+    repo_ref: repoRef
+  })
+
+  const { data: logs } = useViewLogsQuery({
+    pipeline_identifier: pipelineIdentifier,
+    execution_number: executionNum,
+    repo_ref: repoRef,
+    stage_number: String(stageNum),
+    step_number: String(stepNum)
+  })
+
+  useEffect(() => {
+    if (execution?.stages && execution.stages.length > 0) {
+      const stageIdx = stageNum > 0 ? stageNum - 1 : 0
+      setStage(execution.stages[stageIdx])
+    }
+  }, [execution?.stages, stageNum])
```

Comment:
could we use a useMemo for less rerenders?

Reply:
Sure, will fix in a follow up PR.